### PR TITLE
Week 7. Instagram Challenge

### PR DESCRIPTION
**Pull request by yevdyko**

### CODE REVIEW

File: app/helpers/comments_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (k0zakinio) [inline app/helpers/comments_helper.rb]: could delete this

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module ApplicationHelper
```
Comment (k0zakinio) [inline app/helpers/application_helper.rb]: could delete this

File: app/views/comments/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<%= form_for [@post, @comment] do |f| %>
```
Comment (k0zakinio) [inline app/views/comments/new.html.erb]: try not to use instance variables

File: app/views/posts/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<h2>Edit Post</h2>
+<%= form_for @post, html: { multipart: true } do |f| %>
```
Comment (k0zakinio) [inline app/views/posts/edit.html.erb]: try not to use instance variables

File: app/views/posts/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<h2>New Post</h2>
+<%= form_for @post, html: { multipart: true } do |f| %>
```
Comment (k0zakinio) [inline app/views/posts/new.html.erb]: try not to use instance variables

**Comment by tansaku** [comment]
You had your instagram_challenge reviewed by **Evgenii Leshchenko**.
### The good points are:
* Consistent indentation, Dead code and comments removed
* Text expectations appropriate, Test descriptions consistent
* Models appropriately tested
* Models appropriately connected
* Feature tests use 'within' appropriately

### You should consider the following improvements:
* Installation/Usage Instructions should be in README
* Ensure Rakefile has Appropriate Tasks
* Gemfile should use Test Groups
* Ensure spec_helper.rb is Configured Correctly
* Ensure Asset Routes are Set Correctly
* Feature tests should be declarative, not imperative
* Delete empty helper files
* Use nested forms where appropriate
* Avoid generating unnecessary routes using resources in routes.rb
* Misplaced logic in your controllers
* No need to set instance vars when redirecting
* Avoid Duplicated Model Logic
* Differentiate between relative and absolute paths
* Tests should separate out expectations where appropriate
* Use Asset Path correctly for Responsive Images
* Use partials appropriately
* Replace instance variables with local variables
* Avoid Law of Demeter Violations
* Prefer other Semantic HTML Elements to Divs Where Possible
* Avoid anonymous javascript functions

### Additional comments:
1.User can also delete pictures and comments.
2.There is menu of authentication.
3. Authentication is useless because you can do everything without authentication.
4.20 tests, 19 tests work (in the test : line 77 , "test.jpg" not "test.jpeg" ).
5.it used shoulda to test models.
6.Post and comments tests might be separated.
7. In comments and posts controllers may not use instant vars (in methods with redirecting).


see https://github.com/makersacademy/instagram_challenge/blob/master/docs/review.md for more details on the items above

**Pull request closed without merging**